frontend/src/pages/GranteeRecord/pages/TTAHistory.js

```diff
@@ -10,44 +10,32 @@ import Overview from '../../../widgets/DashboardOverview';
 import FilterMenu from '../../../components/filter/FilterMenu';
 import FilterPills from '../../../components/filter/FilterPills';
 import TargetPopulationsTable from '../../../widgets/TargetPopulationsTable';
+import { expandFilters } from '../../../utils';
+
 import './TTAHistory.css';
+import useUrlFilters from '../../../hooks/useUrlFilters';
 
 const defaultDate = formatDateRange({
   yearToDate: true,
   forDateTime: true,
 });
 
-function expandFilters(filters) {
-  const arr = [];
-
-  filters.forEach((filter) => {
-    const { topic, query, condition } = filter;
-    if (Array.isArray(query)) {
-      query.forEach((q) => {
-        arr.push({
-          topic,
-          condition,
-          query: q,
-        });
-      });
-    } else {
-      arr.push(filter);
-    }
-  });
-
-  return arr;
-}
-
 export default function TTAHistory({
   granteeName, granteeId, regionId,
 }) {
-  const [filters, setFilters] = useState([
+  const [filters, setFilters] = useUrlFilters([
     {
       id: uuidv4(),
       topic: 'startDate',
       condition: 'Is within',
       query: defaultDate,
     },
+    {
```

Comment:
Should region be included in the filters ~line 46 since it can't be changed in the UI?

Reply:
Related to my comment above, right now I've got filters that are applied to the widgets, which are different then the filters that are applied to the URL, which are different then are what's editable in the UI. It's a mess. Spending some time now pondering.